static/src/javascripts/projects/commercial/modules/dfp/render-advert.ts

```diff
@@ -151,6 +152,37 @@ const addContentClass = (adSlotNode: HTMLElement) => {
 	}
 };
 
+/**
+ * Avoid CLS when an advert is refreshed, by setting the
+ * min-height of the ad slot to the height of the ad.
+ *
+ * Doesn't work for fluid ads, because we don't know the height of a fluid ad at this point.
+ */
+const setMinHeightOfAdSlot = (advert: Advert): void => {
+	if (
+		getBreakpoint(getViewport().width) === 'mobile' ||
```

Comment:
Unfortunately `getViewport` uses `window.innerX` methods which causes expensive [browser reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a).

This could be costly in the main loop of rendering adverts.

Could we:
- invert the conditional so we only proceed if this is a top-above-nav
- reorder the conditions so the viewport check is last

I imagine we're using getViewport() in lots of places. Ideally we should only read this once and set it in a store to read and update on a resize. But that's outside of the scope of this ticket :)

Reply:
Great points! 😍 

Perhaps this could also be lifted up to [`@guardian/libs`](https://github.com/guardian/csnx/tree/main/libs/%40guardian/libs) as it looks super useful – again, outside this ticket 🙂

Reply:
I'm going to remove the viewport check because it isn't essential. Instead, I'll use the list of desktop ad sizes imported from commercial core to check whether the ad should have a min-height attached to it.